token-erc-1155/chaincode-go/chaincode/contract.go

```diff
@@ -594,6 +757,84 @@ func (s *SmartContract) BroadcastTokenExistance(ctx contractapi.TransactionConte
 	return emitTransferSingle(ctx, transferSingleEvent)
 }
 
+// Name returns a descriptive name for fungible tokens in this contract
+// returns {String} Returns the name of the token
+
+func (s *SmartContract) Name(ctx contractapi.TransactionContextInterface) (string, error) {
+
+	//check if contract has been intilized first
+	initialized, err := checkInitialized(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to check if contract ia already initialized: %v", err)
+	}
+	if !initialized {
+		return "", fmt.Errorf("Contract options need to be set before calling any function, call Initialize() to initialize contract")
+	}
+
+	bytes, err := ctx.GetStub().GetState(nameKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to get Name bytes: %s", err)
+	}
+
+	return string(bytes), nil
+}
+
+// Symbol returns an abbreviated name for fungible tokens in this contract.
+// returns {String} Returns the symbol of the token
+
+func (s *SmartContract) Symbol(ctx contractapi.TransactionContextInterface) (string, error) {
+
+	//check if contract has been intilized first
+	initialized, err := checkInitialized(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to check if contract ia already initialized: %v", err)
+	}
+	if !initialized {
+		return "", fmt.Errorf("Contract options need to be set before calling any function, call Initialize() to initialize contract")
+	}
+
+	bytes, err := ctx.GetStub().GetState(symbolKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to get Symbol: %v", err)
+	}
+
+	return string(bytes), nil
+}
+
+// Set information for a token and intialize contract.
+// param {String} name The name of the token
+// param {String} symbol The symbol of the token
+func (s *SmartContract) Initialize(ctx contractapi.TransactionContextInterface, name string, symbol string) (bool, error) {
+
+	// Check minter authorization - this sample assumes Org1 is the central banker with privilege to intitialize contract
+	clientMSPID, err := ctx.GetClientIdentity().GetMSPID()
+	if err != nil {
+		return false, fmt.Errorf("failed to get MSPID: %v", err)
+	} else if clientMSPID != "Org1MSP" {
```

Comment:
In idiomatic Go code we try to minimize indents.
In this case we don't need an "else if", if there was an error we would have already returned. So we can just do all the checks without nesting. This makes it easier to scan the code by just moving your eyes down vertically.

Same applies in several files in this PR.